Contribution Blog (#519)
3mdeb/news-and-ideas: blog/content/post/2023-04-27-3mdeb-open-source-contribution.md

---

> title: 3mdeb open-source contributions

Reviewer: - We need a better title. This title will give a terrible SEO rank.
- What title would the next post about the 3mdeb contribution be? The same?

Author: @pietrushnic I changed title. Next title will be iteration of this title.

abstract: 'We are excited to share with you the open-source contributions
           of our team. We have been actively contributing to a number of
           open-source projects that have a significant impact on the industry.
           Through these contributions, we have been able to help improve the
           overall functionality and security of these projects, while also
           gaining valuable experience and knowledge that we can apply to our

>            client work.'

Author: The abstract has changed. It now focuses more on keywords and could be easily translated into a social media message.

cover: /covers/image-file.png
author: norbert.kaminski
layout: post
published: true
date: 2023-04-27
archives: "2023"

tags:
  - contribution
  - coreboot
  - fwupd

>   - trenchboot

Author: Tags added.

categories:
  - Manufacturing

---

We are excited to share with you the open-source contributions of our team.
We have been actively contributing to a number of open-source projects that have
a significant impact on the industry. Through these contributions, we have been
able to help improve the overall functionality and security of these projects,
while also gaining valuable experience and knowledge that we can apply to our

> client work.

Reviewer: Abstract repeated, why?

Please reflect on the 3mdeb situation during the contribution period you want to discuss.

Author: https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR30

In addition to contributing to these open-source projects, our team is also
passionate about promoting the benefits of open-source software. We believe
that open-source software offers numerous advantages over proprietary software,
such as greater transparency, flexibility, and security. By contributing to
these projects and advocating for open-source software, we aim to foster
a culture of collaboration, innovation, and inclusivity in the technology
industry. We are committed to making a positive impact on the community
through our work, and we look forward to continuing our open-source

> contributions in the future.

Reviewer: This is trivia for anyone who would like to read about our contribution. Let's respect the reader and not waste one's time.

Let's try to go straight to the point by telling a little about technologies we worked trying to master during the described period. What do those technologies mean for our customers? Why do they need those?

Author: This paragraph is trivial for our core audience, but we want to broaden the circle of potential readers hence the need for general information. I have added descriptions of changes that actually took place in the code.

If we want this blog to be aimed mainly at technical people then I will remove the sections that explain what each project does and the general concept of working on open-source projects

Reviewer: With an additional introduction, this paragraph is acceptable.

## Our contributions:

### coreboot

coreboot is an open-source firmware that provides a lightweight, secure,
and fast boot experience for PCs, laptops, servers, and embedded devices.
If you are interested in improving the security of your devices while reducing
your dependence on proprietary firmware, you could benefit from using coreboot.
We can also help you reduce time-to-market by simplifying the firmware

> development process.

Reviewer: Again trivia, more respect to the reader.

<!-- List of contribution-->

### fwupd

fwupd is an open-source daemon that manages the firmware updates of various
devices. You will surely benefit from using fwupd if you are interested in
automating the firmware update process on your devices and reducing the risk of
security vulnerabilities. fwupd is compatible with a wide range of devices,
including laptops, desktops, and IoT devices. Our team can help you seamlessly
integrate into the fwupd ecosystem across a variety of platforms and operating
systems. Say goodbye to clunky update processes and hello to a streamlined,

> hassle-free experience with fwupd!

Reviewer: a/a - you can always link to the project webpage/documentation to follow the DRY rule.

meets your unique requirements and security needs, Yocto is the open-source
project for you. As a comprehensive suite of tools and templates, Yocto
provides the flexibility and customization you need to create a bespoke
solution for your device.

By partnering with our team, we can help you leverage the power of Yocto and
build a custom Linux distribution that fully aligns with your vision. From
feature-rich IoT devices to mission-critical servers, we've got you covered.
Let us create a personalized solution that meets your exact specifications
and takes your device's capabilities to the next level.

<!-- List of contribution-->

### Trenchboot

Trenchboot is a game-changing open-source project that provides a powerful set
of tools for measuring and attesting to the trustworthiness of a system during
the boot process. If you're looking for secure boot processes for your devices
- be it IoT devices, servers, or laptops - Trenchboot is the solution you've
been searching for. Our team is equipped to enable Trenchboot for you so that
you can measure and attest to the integrity of your system's boot process,
and enjoy enhanced protection against attacks and device security breaches.

> Let us help you safeguard your devices with Trenchboot.

Author: Description changed, now it's based on the TrenchBoot documentation

<!-- List of contribution-->

## Summary

> If you're looking for expert guidance on open-source projects such as Coreboot,

Reviewer: ```suggestion
If you're looking for expert guidance on open-source projects such as coreboot,
```

fwupd, Yocto, and Trenchboot, our team is here to help. We are committed to
helping our clients achieve their security and customization goals through
our open-source expertise, and we invite you to join us on this journey.
[We'd love to discuss the details](https://3mdeb.com/contact/)

> of how we can work together to bring your project to the next level.

Reviewer: We need more meat than that. Why may someone need those technologies?

If you are passionate about these topics, we also welcome you to join our
recruitment process and become a part of our team.

> [Check here](https://3mdeb.com/careers/) possible career paths.

Reviewer: Let's start each paragraph with an answer to why. Why should someone keep reading? Why this paragraph is interesting to me? What goal you can help me achieve if I will waste time reading this?

Additionally, we invite everyone to join our community and
[matrix space](https://matrix.to/#/%23dasharo-osf-vpub:matrix.org),
where you can connect with like-minded individuals and stay up-to-date with the
latest news in the open-source world.

> Don't forget to mark your calendars for

Reviewer: I like what you wrote in https://blog.3mdeb.com/2023/2023-01-30-3mdeb-insights-winter-2023/

In paragraphs:
- Dasharo User Group (DUG) `#1`
- Dasharo Developers vPub 0x6

Why not have the same two paragraphs but for `#2` and 0x6 with a small summary of the previous DUG and some inspiration for what can happen next time?

Reviewer: One paragraph of summary of DUG#1 and vPub 0x6 would be appreciated.

Author: I added summary 8e27f10b6db61e09c74d621776c66c05c8d031ad

---
title: "3mdeb Insights: A Look into 3mdeb's Open-Source Contributions"

> abstract: "We're thrilled to share our team's remarkable open-source

Reviewer: ```suggestion
abstract: "We're thrilled to share our team's summary of open-source
```

           contributions from 2021 to the first half of 2023 with you.
           We've actively worked on enhancing various projects, including

>            coreboot and fwupd, leaving a significant impact on the industry.

Reviewer: ```suggestion
           coreboot and fwupd.
```

Please correct wrapping

           Our involvement has not only boosted the functionality and security
           of these projects but also equipped us with valuable expertise for
           our clients. Stay tuned for an insightful blog post diving into our

>            open-source contributions!"

Reviewer: Please rewrite based on following example: https://github.com/3mdeb/news-and-ideas/blob/master/blog/content/post/2023-05-18-our_contribution_to_coreboot_4_20_release.md?plain=1#L3

Recently, our engineers have focused on expanding coreboot support for the
latest platforms, it is worth mentioning that Karol Zmyslowski has added
[support for Jasper Lake](https://review.coreboot.org/c/coreboot/+/73934).

Michał Kopeć and Michał Żygowski had a major impact on Alder Lake support in
coreboot. Michał Kopeć has added [support for Alder Lake](https://review.coreboot.org/c/coreboot/+/63374)
and Michał Żygowski in the [set of patches](https://review.coreboot.org/c/coreboot/+/68449/11)
has added improvements huge improvements. Also, Michał Żygowski made a significant
contribution to [Tiger Lake support](https://review.coreboot.org/c/coreboot/+/56171).

In the previous year, our team also added
[support for MSI](https://review.coreboot.org/c/coreboot/+/68078) PRO Z690-A
WIFI and MSI PRO Z690-A in DDR4 and DDR5 versions.

Krystian Hebel, Sergii Dmytruk, and Yaroslav Kurlaev have made major contributions
to developing and maintaining support for Power platforms. Their changes include
[emulation improvements](https://review.coreboot.org/c/coreboot/+/67061),
and fixing ongoing issues related to these platforms.

> #### Platform security (TPM, TXT)

Reviewer: ```suggestion
#### Platform security (TPM, Intel TXT)
```

Please fix also other instances

Author: Deleted this section and now we're linking to the 4.20 release blog. https://github.com/3mdeb/news-and-ideas/commit/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

Our team also worked on [TPM](https://3mdeb.com/shop/modules/tpm-2-0/) and
[TXT](https://blog.3mdeb.com/2022/2022-03-17-optiplex-txt/) solutions that
could support the security of coreboot-based platforms. Sergii Dmytruk added
quite a few improvements related to
[TPM support](https://review.coreboot.org/c/coreboot/+/68748),
while Michal Zygowski added some
[TXT improvements](https://review.coreboot.org/c/coreboot/+/59519).

#### coreboot documentation improvements

> The work of our engineers was not closed only to changes in the code.

Reviewer: I'm not sure if this is English: closed only to changes, what you mean hear? Maybe this is translation from Polish?

Author: Deleted this section and now we're linking to the 4.20 release blog. https://github.com/3mdeb/news-and-ideas/commit/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

It is equally important to make the code readable, clear, and understandable
to everyone. Hence, it is worth highlighting our efforts to keep the coreboot
documentation up to date. In this matter, the main contributors were
[Michał Żygowski](https://review.coreboot.org/c/coreboot/+/65702) and

> [Sergii Dmytruk](https://review.coreboot.org/c/coreboot/+/68752)

Reviewer: It is better to provide the title or name of the change linked to Gerrit and then mention who did that. Linking the name to change is weird and unexpected.

Author: Deleted this section and now we're linking to the 4.20 release blog. https://github.com/3mdeb/news-and-ideas/commit/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

If you are interested in improving the security of your devices while reducing
your dependence on proprietary firmware, you could benefit from using coreboot.
We can also help you reduce time-to-market by simplifying the firmware
development process. As well as if you want your firmware to be user-friendly
and well-documented user-friendly and well-documented then coreboot is for you.

---

![fwupd-logo](/img/fwupd-logo.svg)

[fwupd](https://3mdeb.com/shop/services/basic-firmware-update-integration/)
is an open-source daemon that manages the firmware updates of various
devices. You will surely benefit from using fwupd if you are interested in
automating the firmware update process on your devices and reducing the risk of
security vulnerabilities. fwupd is compatible with a wide range of devices,
including laptops, desktops, and IoT devices.

Changes to fwupd can be divided into several areas. The first is the addition of
[support for Qubes OS](https://github.com/fwupd/fwupd/pull/2710/commits/295418ef784408503b6a62991a74aa4bc886c4e0).

> These changes allow firmware updates from within one of the most

Reviewer: ```suggestion
These changes allow firmware updates from within reasonably
```

Author: Fixed https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR105

> These changes allow firmware updates from within one of the most

Author: Fixed https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR105

> secure operating systems. Norbert Kamiński was responsible for these changes.

Reviewer: Link to profile.

The whole paragraph has a very generic description. It could be better and contain more meat, especially since you are the author of the changes.

The second significant set of changes was the addition of
[support for FreeBSD](https://github.com/fwupd/fwupd/pull/3330). These changes
made it possible to open fwupd to another group of operating systems.
These changes were worked on by Michal Kopeć, Sergii Dmytruk, and Norbert Kamiński.

Another sizable change that our engineers were responsible for was extending
[flashrom support](https://github.com/fwupd/fwupd/commit/2bac03eee1a68becc96712d3a86656cc2e8fa19d)

> and adding support for Tuxedo laptops.

Reviewer: Link to TUXEDO and maybe Dasharo Universe? How this change can help other customers?

Author: Link and description added https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR118

Our team can help you seamlessly integrate into the fwupd ecosystem across
a variety of platforms and operating systems. Say goodbye to clunky update
processes and hello to a streamlined, hassle-free experience with fwupd!

---

![yocto logo](/img/YoctoProject_Logo_RGB.jpg)

If you're looking for a tailored Linux-based operating system that perfectly
meets your unique requirements and security needs, Yocto is the open-source
project for you. As a comprehensive suite of tools and templates, Yocto
provides the flexibility and customization you need to create a bespoke
solution for your device.

> Our engineers have added some important fixes to some of Yocto's most popular

Reviewer: some some

Author: fixed https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR201

> layers. Tomasz Żyjewski added support for the Dunfell version of

Reviewer: Profile link?

Author: Added https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR208

[meta-openwrt](https://github.com/kraj/meta-openwrt/pull/113)
as well as added support for
[python3-binwalk](https://github.com/openembedded/meta-openembedded/commit/7ea0e04aaee19f61e18bf998bb07f02e52d8146e)
and [python3-uefi-firmware](https://github.com/openembedded/meta-openembedded/commit/3e70428db7b48573883aa50de636d93757dd263e)

> in the `meta-openembedded` layer.

Reviewer: How do those changes help? OpenWRT support was significant. There was even a conference presentation associated.

Author: Description added https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR205

the [meta-sunxi](https://github.com/linux-sunxi/meta-sunxi/pull/256) layer.

By partnering with our team, we can help you leverage the power of Yocto and
build a custom Linux distribution that fully aligns with your vision. From
feature-rich IoT devices to mission-critical servers, we've got you covered.
Let us create a personalized solution that meets your exact specifications
and takes your device's capabilities to the next level.

![Trenchboot logo](/img/trenchboot_logo.svg)

[TrenchBoot](https://blog.3mdeb.com/tags/trenchboot/) is a framework that
allows individuals and projects to build security engines to perform launch
integrity actions for their systems. The framework builds upon Boot Integrity
Technologies (BITs) that establish one or more Roots of Trust (RoT) from which
a degree of confidence that integrity actions were not subverted is derived.

The biggest changes took place in the landing-zone component. Worth mentioning
is the addition of [support for the Xen hypervisor](https://github.com/TrenchBoot/landing-zone/pull/60)
and [multiboot2 support for the GRUB2 bootloader](https://github.com/TrenchBoot/landing-zone/pull/64).

> The author of these changes is Krystian Hebel.

Reviewer: Link profile.

Author: Link added https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR311

If you're looking for methods of enhancing boot security for your devices,
Trenchboot is the solution you've been searching for. Our team is equipped
to enable Trenchboot for you so that you can enjoy enhanced protection against
attacks and device security breaches. Let us help you safeguard your devices

> with Trenchboot.

Reviewer: Explaining how it aligns with our mission of improving the trustworthiness of every computing device would be appreciated.

Author: Added https://github.com/3mdeb/news-and-ideas/commit/709f02b376e271cc6347a0600a13a0206b46704e#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR315

## Upcoming events

Don't forget to mark your calendars for
[Dasharo User Group #2](https://vpub.dasharo.com/e/7/dasharo-user-group-2),
which will take place on July 6th, 2023. This is a great opportunity to learn

> more about open-source projects and connect with other members of the community.

Reviewer: As well as discuss the 3mdeb open-source contribution.

Author: Added https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR340

The Dasharo User Group (DUG) is an important forum for users of Dasharo to
come together, share their knowledge, and stay informed about the latest
developments in the Dasharo ecosystem. The DUG is a platform for users to
connect and learn about new features and updates that are coming to Dasharo.
The first DUG event will take place in early March and will include a variety
of discussions on different topics related to Dasharo. The agenda for the event
will be shared in the next month. The event will be a great opportunity for
Dasharo users to meet other users, learn new things, and share their
knowledge and experience with others.

Dasharo vPub 0x7 is a follow-up event to DUG#2 and will provide a space
to engage in more informal conversations and discussions that may not have
been covered during DUG#2. The vPub is designed to be a less structured, more
relaxed environment where the community can discuss topics that are of interest
to them. This can include off-topic discussions, technical challenges they are

> facing, and ideas for new features or improvements.

Reviewer: Why not link information about previous events, videos from `DUG#1`, and agenda from vPub 0x6?

Author: Links added. https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR358

## Summary

These are just a selection of our contributions to open-source. Since its
inception, 3mdeb has contributed changes to more than 100,000 lines of code
in open-source projects. So if you're looking for expert guidance on open-source
projects such as coreboot, fwupd, Yocto, and Trenchboot, our team is here to
help. We are committed to helping our clients achieve their security and
customization goals through our open-source expertise, and we invite you to

> join us on this journey.

Reviewer: Too much `help` word

Author: Section changed. https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR369

The first DUG event will take place in early March and will include a variety
of discussions on different topics related to Dasharo. The agenda for the event
will be shared in the next month. The event will be a great opportunity for
Dasharo users to meet other users, learn new things, and share their
knowledge and experience with others.

Dasharo vPub 0x7 is a follow-up event to DUG#2 and will provide a space
to engage in more informal conversations and discussions that may not have
been covered during DUG#2. The vPub is designed to be a less structured, more
relaxed environment where the community can discuss topics that are of interest
to them. This can include off-topic discussions, technical challenges they are
facing, and ideas for new features or improvements.

The Dasharo User Group (DUG#1) and vPub 0x6 event achieved great success,
offering insightful presentations and engaging discussions on topics related
to open-source firmware, hardware, and security. Key highlights encompassed
Dasharo's roadmap, the groundbreaking potential of NovaCustom's hardware
and open-source firmware for enhancing the laptop experience, the Dasharo Tool

> Suite roadmap, notable Dasharo features like X11 and RPL-S CPU support,

Reviewer: ```suggestion
Suite roadmap, notable Dasharo Community Support ports like Supermciro X11SSH support and RPL-S CPU,
```

---

From 2021 to 2022, our team of developers thrived, making significant
contributions to firmware projects. Our primary focus revolved around

> [coreboot](https://3mdeb.com/training/#coreboot-for-embedded-linux-developers),

Reviewer: Why are we linking to training? We can connect to the official coreboot page or tag coreboot in our blog post.

Linking to training does not make sense in this context.

Reviewer: Each fix should be addressed in a separate commit; otherwise, realizing what is fixed for this particular issue and what is fixed for others is still problematic.

Please stop making bulk commits during the review.

a firmware solution that captured our attention. We also dedicated efforts to

> the advancement of fwupd and flashrom firmware update systems. But our

Reviewer: Why not link to the fwupd and flashrom project pages, if we don't have associated tags in blog

> dedication to system security didn't stop there—we actively developed

Reviewer: ```suggestion
dedication to system security didn't stop there - we actively developed
```

through our work, and we look forward to continuing our open-source
contributions in the future.

## Our contributions

![coreboot logo](/covers/coreboot-logo.svg)

> [coreboot](https://3mdeb.com/training/#coreboot-for-embedded-linux-developers)

Reviewer: here I would link to coreboot project page

is an open-source firmware that provides a lightweight, secure,
and fast boot experience for PCs, laptops, servers, and embedded devices.
The changes made by our engineers can be divided into three main groups.

#### Adding support for new platforms and maintaining existing ones

Recently, our engineers have focused on expanding coreboot support for the
latest platforms, it is worth mentioning that Karol Zmyslowski has added

> [support for Jasper Lake](https://review.coreboot.org/c/coreboot/+/73934).

Reviewer: Let's add `Intel` for every microarchitecture name.

Michał Kopeć and Michał Żygowski had a major impact on Alder Lake support in
coreboot. Michał Kopeć has added [support for Alder Lake](https://review.coreboot.org/c/coreboot/+/63374)
and Michał Żygowski in the [set of patches](https://review.coreboot.org/c/coreboot/+/68449/11)

> has added improvements huge improvements. Also, Michał Żygowski made a significant

Reviewer: `has added improvements huge improvements` - we have to be more precise, Michał added support that enables DMA protection, which is very important to protect against [DMA attacks](https://en.wikipedia.org/wiki/DMA_attack) at the firmware level. He also extended previously contributed [support for HSPHY](https://review.coreboot.org/plugins/gitiles/coreboot/+/9b0f169d2503dc044500e2790b987d766928782a) by making it work with DMA protection enabled.

contribution to [Tiger Lake support](https://review.coreboot.org/c/coreboot/+/56171).

> In the previous year, our team also added

Reviewer: Previous to which, do the last paragraphs are chronological? Those instead look like reverse chronology.

Consistency is essential, whether you use chronological or reverse order.

[support for MSI](https://review.coreboot.org/c/coreboot/+/68078) PRO Z690-A
WIFI and MSI PRO Z690-A in DDR4 and DDR5 versions.

> Krystian Hebel, Sergii Dmytruk, and Yaroslav Kurlaev have made major contributions

Reviewer: @Asiderr Yaroslav is not and was not a 3mdeb employee, so I'm sure why you referred him.

Author: He was in the spreadsheet, deleted. https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

> to developing and maintaining support for Power platforms. Their changes include

Reviewer: ```suggestion
to develop and maintain support for IBM POWER8 and POWER9 platforms. Their changes include
```

Please try to use Grammarly.

Author: Used grammarly (now and before). https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

[emulation improvements](https://review.coreboot.org/c/coreboot/+/67061),
and fixing ongoing issues related to these platforms.

#### Platform security (TPM, TXT)

Our team also worked on [TPM](https://3mdeb.com/shop/modules/tpm-2-0/) and
[TXT](https://blog.3mdeb.com/2022/2022-03-17-optiplex-txt/) solutions that
could support the security of coreboot-based platforms. Sergii Dmytruk added
quite a few improvements related to
[TPM support](https://review.coreboot.org/c/coreboot/+/68748),
while Michal Zygowski added some

> [TXT improvements](https://review.coreboot.org/c/coreboot/+/59519).

Reviewer: This needs a more in-depth explanation of why is the value of those changes.

Author: Deleted this section and now we're linking to the 4.20 release blog. https://github.com/3mdeb/news-and-ideas/commit/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR79

#### coreboot documentation improvements

The work of our engineers was not closed only to changes in the code.
It is equally important to make the code readable, clear, and understandable
to everyone. Hence, it is worth highlighting our efforts to keep the coreboot
documentation up to date. In this matter, the main contributors were
[Michał Żygowski](https://review.coreboot.org/c/coreboot/+/65702) and
[Sergii Dmytruk](https://review.coreboot.org/c/coreboot/+/68752)

If you are interested in improving the security of your devices while reducing
your dependence on proprietary firmware, you could benefit from using coreboot.
We can also help you reduce time-to-market by simplifying the firmware
development process. As well as if you want your firmware to be user-friendly

> and well-documented user-friendly and well-documented then coreboot is for you.

Reviewer: Instead of this paragraph, I would try to pitch Dasharo, especially since most of the above changes were part of Dasharo's product development.

Author: Pitch added https://github.com/3mdeb/news-and-ideas/pull/519/commits/6232e10165348e655ed155ab352d394464008f15#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR85

---

![fwupd-logo](/img/fwupd-logo.svg)

> [fwupd](https://3mdeb.com/shop/services/basic-firmware-update-integration/)

Reviewer: Link to service in shop should be at the end of the section, here I would link to fwupd/LVFS project to explain what it is.

Author: Moved this link https://github.com/3mdeb/news-and-ideas/pull/519/commits/68a72cdebcee326365faf61a3b7861f718765122#diff-5dff579917e1df023c3d68f093f637f7fb497ee2f22650cc786ac30a195a7eceR123
